languages/swift/exercises/concept/structs-and-classes/.docs/after.md

```diff
@@ -0,0 +1,298 @@
+
+```swift
+struct CharacterStats {
+  var health = 0.0
+  var speed = 0
+  var strength = 0
+
+  mutating func takeHit(_ damage: Double) {
+    health = max(0.0, health - damage)
+  }
+  func canLift(_ weight: Int) -> Bool {
+    weight < strength * 100
+  }
+}
+
+class GameCharacter {
+  var stats = CharacterStats()
+  var characterClass: String?
+  var name: String?
+  var active = false
+  let id: String = makeRandomID()
+
+  func takesDamage(_ damage: Double) {
+    stats.takeHit(damage)
+    if stats.health <= 0 {
+      active = false
+    }
+  }
+  func sayName() -> String {
+    return "My name is \(name ?? "no one"), my class is \(characterClass ?? "undetermined")"
+  }
+  func lift(_ weight: Int) -> String {
+    if stats.canLift(weight) {
+      return "No problem!"
+    } else {
+      return "Ooof! No way."
+    }
+  }
+}
+```
+
+These methods can be called using dot notation, just like properties.
+
+```swift
+var myChar = GameCharacter()
+myChar.stats = CharacterStats(health: 72.8, speed: 19, strength: 6)
+myChar.name = "Shadowfly"
+myChar.characterClass = "spy"
+myChar.active = true
+
+myChar.active
+// => true
+myChar.sayName()
+// => "My name is Shadowfly, my class is spy"
+myChar.lift(750)
+// => "Ooof! No way."
+myChar.takesDamage(80)
+myChar.active
+// => false
+```
+
+### Self
+
+Instances of structs and classes each have an implicit value named `self` which refers to the instance itself. There are multiple uses for `self`, but it is most commonly used to disambiguate the names of properties and methods of the struct/class when there may be some confusion.
+
+```swift
+struct MySelf {
+  var x = 0
+
+  mutating func move(x: Int) {
+    // here if we just say x = x it is unclear if we mean
+    // the property x or the method parameter x, so we use
+    // self for clarity
+    self.x = x
+  }
+}
+```
+
+Structs and classes also have a `Self` value which refers the the type if the struct or class, rather than the instance of the struct or class.
+
+### Value types v. reference types
+
+While Swift have many different types, each of these types fall into one of two different classifications, _value types_ and _reference types_. In Swift, [structs are value types][structs-are-value-types] while [classes are reference types][classes-are-reference-types].
+
+The primary difference between value types and reference types is that, when value types are assigned to a variable or constant or when passed into a function, it is actually a copy of the value that is assigned or passed, not that particular value. Modifying one copy of a struct will not affect other copies. With reference types, a reference to that value is assigned or passed, so two different names can refer to the same instance of a class. This can lead to modification of the original instance when accessed under the different name.
```

Comment:
```suggestion
The primary difference between value types and reference types is that, when value types are assigned to a variable or constant or when passed into a function, it is actually a copy of the value that is assigned or passed, not that particular value. Modifying one copy of a struct will not affect other copies. With reference types, a reference to that value is assigned or passed, so two different names can refer to the same instance of a class. This can lead to modification of the original instance when accessed under a different name.
```

Reply:
Fixed in latest commit